.github/workflows/build-and-test.yml

```
@@ -6,6 +6,7 @@ on:
       - master
     types:
       - opened
+      - synchronize
```

```suggestion
      - synchronize
      - reopened
```

Which is the same as leaving it blank:
https://frontside.com/blog/2020-05-26-github-actions-pull_request/#how-do-workflows-trigger-on-pull_request

---

Not sure I see the point of re-running the test suite upon re-opening a closed PR…

If you re-open the test suite without having pushed any new commits, then the last CI check performed on the last pushed commit before the PR was closed is still valid; and after re-opening, if you change the PR by pushing new commits, the `synchronize` setting will trigger a new run.

The only thing that running the test suite immediately upon re-opening the PR would test for is the following scenario:

* A PR is submitted while the master branch is in state _t0_. Let’s say the CI checks pass.
* The PR is closed for whatever reasons.
* Other, unrelated changes happen on the master branch, which is now in state _t1_.
* Later, the PR is re-openened. The PR still applies cleanly against master’s _t1_ state, but somehow, because of the changes between _t0_ and _t1_, the PR now fails the CI checks.

This is no different from the sitation where a PR passes the CI checks at a given moment, but then is left (without being closed) without being merged for a while, and then is finally approved and merged – by the time the PR is merged, maybe the changes that happened in the meantime on the master branch would actually cause the PR to fail, but we do not test that.

---

Ok! Lets follow your lead!